dev/com.ibm.ws.config/src/com/ibm/ws/config/admin/internal/ConfigurationStorageHelper.java

```diff
@@ -0,0 +1,673 @@
+        for (int i = 0; i < size; i++) {
+            result[i] = dis.readInt();
+        }
+        return result;
+    }
+
+    private static Object readIntegerArray(DataInputStream dis, int size) throws IOException {
+        Integer[] result = new Integer[size];
+        for (int i = 0; i < size; i++) {
+            result[i] = dis.readInt();
+        }
+        return result;
+    }
+
+    private static Object readStringArray(DataInputStream dis, int size) throws IOException {
+        String[] result = new String[size];
+        for (int i = 0; i < size; i++) {
+            result[i] = dis.readUTF();
+        }
+        return result;
+    }
+
+    private static Object readProtectedStringArray(DataInputStream dis, int size) throws IOException {
+        SerializableProtectedString[] result = new SerializableProtectedString[size];
+        for (int i = 0; i < size; i++) {
+            result[i] = new SerializableProtectedString(dis.readUTF().toCharArray());
+        }
+        return result;
+    }
+
+    private static Object readOnErrorArray(DataInputStream dis, int size) throws IOException {
+        OnError[] result = new OnError[size];
+        OnError[] values = OnError.values();
+        for (int i = 0; i < size; i++) {
+            result[i] = values[dis.readInt()];
+        }
+        return result;
+    }
+
+    private static Object readPrimative(DataInputStream dis, int type) throws IOException {
+        Object value;
+        if (type == BYTE) {
+            value = dis.readByte();
+        } else if (type == SHORT) {
+            value = dis.readShort();
+        } else if (type == BOOLEAN) {
+            value = dis.readBoolean();
+        } else if (type == CHAR) {
+            value = dis.readChar();
+        } else if (type == DOUBLE) {
+            value = dis.readDouble();
+        } else if (type == LONG) {
+            value = dis.readLong();
+        } else if (type == FLOAT) {
+            value = dis.readFloat();
+        } else if (type == INTEGER) {
+            value = dis.readInt();
+        } else if (type == STRING) {
+            value = dis.readUTF();
+        } else if (type == PROTECTED_STRING) {
+            value = new SerializableProtectedString(dis.readUTF().toCharArray());
+        } else if (type == ONERROR) {
+            int ordinal = dis.readInt();
+            value = OnError.values()[ordinal];
+        } else {
+            value = null;
+        }
+
+        return value;
+    }
+
+    private static Collection<?> readCollection(DataInputStream dis) throws IOException {
+        List<Object> result = new ArrayList<>();
+
+        int len = dis.readInt();
+        if (len > 0) {
+            int type = dis.readByte();
+            for (int i = 0; i < len; i++) {
+                result.add(readPrimative(dis, type));
+            }
+        }
+        return result;
+    }
+
+    public static void writeMap(DataOutputStream dos, MapIterable map) throws IOException {
```

Comment:
does this need to be public?

Reply:
Yes for usage by the variable registry cache